### PR TITLE
feat(meta): let `pin_version` use push scheme

### PR DESCRIPTION
**Pull request by soundOfDestiny**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

let `pin_version` use push scheme

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

### CODE REVIEW

File: src/meta/src/hummock/hummock_manager.rs [hunk 1]
```diff
@@ -494,6 +479,36 @@ where
...
+        {
+            drop(versioning_guard);
+            self.check_state_consistency().await;
+        }
+
+        Ok(())
+    }
+
     #[named]
     pub async fn unpin_version(
```
Comment (zwang28) [inline src/meta/src/hummock/hummock_manager.rs:513]: unpin_version will break the new assumption "version_pin.version_id is not empty". What about remove unpin_version, change version_pin.version_id to a single value and rename it to something like min_pinned_id?

File: src/meta/src/hummock/hummock_manager.rs [hunk 2]
```diff
@@ -497,24 +480,44 @@ where
     }
 
     #[named]
-    pub async fn unpin_version(
+    pub async fn unpin_version_before(
```
Comment (zwang28) [inline src/meta/src/hummock/hummock_manager.rs:483]: Would you please add doc for this method?

File: src/meta/src/hummock/hummock_manager.rs [hunk 3]
```diff
@@ -497,24 +480,44 @@ where
...
+        {
+            drop(versioning_guard);
+            self.check_state_consistency().await;
+        }
+
+        Ok(())
+    }
+
+    #[named]
+    pub async fn unpin_version(&self, context_id: HummockContextId) -> Result<()> {
```
Comment (zwang28) [inline src/meta/src/hummock/hummock_manager.rs:513]: ditto

File: src/meta/src/hummock/hummock_manager_tests.rs [hunk 4]
```diff
@@ -72,20 +70,11 @@ async fn test_hummock_pin_unpin() {
         let pinned_versions = HummockPinnedVersion::list(env.meta_store()).await.unwrap();
         assert_eq!(pin_versions_sum(&pinned_versions), 1);
         assert_eq!(pinned_versions[0].context_id, context_id);
-        assert_eq!(pinned_versions[0].version_id.len(), 1);
-        assert_eq!(pinned_versions[0].version_id[0], version_id);
     }
 
     // unpin nonexistent target will not return error
```
Comment (zwang28) [inline src/meta/src/hummock/hummock_manager_tests.rs:75]: please help this fix comment (Line 75) because meaning of unpin_version has changed.

**Review by zwang28 — APPROVED** [review]
LGTM

File: src/meta/src/hummock/hummock_manager.rs (hunk 2 above)
Comment (soundOfDestiny) [reply src/meta/src/hummock/hummock_manager.rs:482]: > Would you please add doc for this method?

added

File: src/meta/src/hummock/hummock_manager.rs (hunk 3 above)
Comment (soundOfDestiny) [reply src/meta/src/hummock/hummock_manager.rs:513]: > ditto

added

File: src/meta/src/hummock/hummock_manager_tests.rs (hunk 4 above)
Comment (soundOfDestiny) [reply src/meta/src/hummock/hummock_manager_tests.rs]: > please help this fix comment (Line 75) because meaning of unpin_version has changed.

fixed

**Pull request merged**